rollouts/controllers/rollout_controller.go

```diff
@@ -680,13 +707,49 @@ func (r *RolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	if err := r.store.Init(); err != nil {
 		return err
 	}
-	// TODO: watch cluster resources as well
+
+	var containerCluster unstructured.Unstructured
+	containerCluster.SetGroupVersionKind(configConnectorContainerClusterGVK)
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&gitopsv1alpha1.Rollout{}).
 		Owns(&gitopsv1alpha1.RemoteRootSync{}).
+		Watches(
+			&source.Kind{Type: &containerCluster},
+			handler.EnqueueRequestsFromMapFunc(r.mapClusterUpdateToRequest),
+		).
 		Complete(r)
 }
 
+func (r *RolloutReconciler) mapClusterUpdateToRequest(cluster client.Object) []reconcile.Request {
+	logger := log.FromContext(context.Background())
+
+	var requests []reconcile.Request
+
+	allRollouts, err := r.listAllRollouts(context.Background())
+	if err != nil {
+		logger.Error(err, "failed to list rollouts")
+		return []reconcile.Request{}
+	}
+
+	for _, rollout := range allRollouts {
+		selector, err := metav1.LabelSelectorAsSelector(rollout.Spec.Targets.Selector)
+		if err != nil {
+			logger.Error(err, "failed to create label selector", "rolloutName", rollout.Name)
+			continue
+		}
+
+		rolloutDeploysToCluster := rolloutIncludesCluster(rollout, cluster.GetName())
+		clusterInTargetSet := selector.Matches(labels.Set(cluster.GetLabels()))
+
+		if rolloutDeploysToCluster || clusterInTargetSet {
```

Comment:
would be good to add a comment explaining why both needs to be done.

Reply:
Added a comment explaining the rational when a rollout needs to be reconciled for a cluster update.